tarofreighter/chain_porter.go

```diff
@@ -81,6 +81,14 @@ type ChainPorter struct {
 
 	exportReqs chan *AssetParcel
 
+	// subscribers is a map of components that want to be notified on new
+	// events, keyed by their subscription ID.
+	subscribers map[uint64]*chanutils.EventReceiver[Event]
```

Comment:
Can alternatively use/copy/replicate this map type: https://github.com/lightningnetwork/lnd/blob/48c8c1bf48ed9be7a014803ea4c3b4624407dcc3/lntemp/node/sync_map.go#L5

Then this would be:
```go
subscribers SyncMap[uint64, *chanutils.EventReceiver[Event]]
```

Reply:
Also we could alternatively lift out this new state/methods into a new struct, then _embed_ that struct within this one.

Reply:
I've gone down the embedded structure route.